src/main.ts

```diff
@@ -4,13 +4,14 @@ import * as exec from '@actions/exec';
 import * as io from '@actions/io';
 
 import { FormatType, SecretParser } from 'actions-secret-parser';
+import { ServicePrincipalLogin } from './PowerShell/ServicePrincipalLogin';
 
 var azPath: string;
 var prefix = !!process.env.AZURE_HTTP_USER_AGENT ? `${process.env.AZURE_HTTP_USER_AGENT}` : "";
 
 async function main() {
-    try{
-        // Set user agent varable
+    try {

```

Comment:
Can we refactor az cli login to a different file as well. This will improve readability. Also we can add error handling login individually in each class.

Just read input, preferably in a different function which returns a taskparameter object. Invoke both login classes as required. Wrap within try catch and throw generic error. Rest logic can be handled in individual classes.

Reply:
This seems like a good approach, but i think its not within this scope. This can be done later and separately.